File: output/GPT3.5-Turbo/xCodeEval_4000/Java/Python/ff695da93888df1b821f8a34023f3dc1.py
class TrieNode:
    def __init__(self):
        self.children = {}
        self.endOfWord = False

class Trie:
    def __init__(self):
        self.root = TrieNode()

    def insert(self, word):
        current = self.root
        for ch in word:
            if ch not in current.children:
                current.children[ch] = TrieNode()
            current = current.children[ch]
        current.endOfWord = True

    def search(self, word):
        current = self.root
        for ch in word:
            if ch not in current.children:
                return False
            current = current.children[ch]
        return current.endOfWord

    def delete(self, word):
        self._delete(self.root, word, 0)

    def _delete(self, current, word, index):
        if index == len(word):
            if not current.endOfWord:
                return False
            current.endOfWord = False
            return len(current.children) == 0

        ch = word[index]
        if ch not in current.children:
            return False

        shouldDeleteCurrentNode = self._delete(current.children[ch], word, index + 1)

        if shouldDeleteCurrentNode:
            del current.children[ch]
            return len(current.children) == 0

        return False

File: output/GPT3.5-Turbo/xCodeEval_remove_4000/Java/C/ff695da93888df1b821f8a34023f3dc1.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>
#include <math.h>

struct TrieNode {
    struct TrieNode* children[26];
    bool isEndOfWord;
};

struct TrieNode* createNode() {
    struct TrieNode* node = (struct TrieNode*)malloc(sizeof(struct TrieNode));
    node->isEndOfWord = false;
    for (int i = 0; i < 26; i++) {
        node->children[i] = NULL;
    }
    return node;
}

void insert(struct TrieNode* root, char* word) {
    struct TrieNode* current = root;
    for (int i = 0; i < strlen(word); i++) {
        int index = word[i] - 'a';
        if (current->children[index] == NULL) {
            current->children[index] = createNode();
        }
        current = current->children[index];
    }
    current->isEndOfWord = true;
}

bool search(struct TrieNode* root, char* word) {
    struct TrieNode* current = root;
    for (int i = 0; i < strlen(word); i++) {
        int index = word[i] - 'a';
        if (current->children[index] == NULL) {
            return false;
        }
        current = current->children[index];
    }
    return current->isEndOfWord;
}

void delete(struct TrieNode* root, char* word) {
    struct TrieNode* current = root;
    for (int i = 0; i < strlen(word); i++) {
        int index = word[i] - 'a';
        if (current->children[index] == NULL) {
            return;
        }
        current = current->children[index];
    }
    current->isEndOfWord = false;
}

int main() {
    struct TrieNode* root = createNode();

    insert(root, "hello");
    insert(root, "world");

    printf("%d\n", search(root, "hello")); // Output: 1
    printf("%d\n", search(root, "world")); // Output: 1
    printf("%d\n", search(root, "hi")); // Output: 0

    delete(root, "hello");

    printf("%d\n", search(root, "hello")); // Output: 0

    return 0;
}
